File: Higher_Order_Functions/PartialFunctions/SendMessage.scala

def sendMessage(from: String, to: String, text: String): String = '@' + to + ':' + text + '-' + from;

val hiMessage = sendMessage(_:String,_:String,"Hi to all");
val hiMessageToDAP = hiMessage(_:String,"DAP Team");
val hiMessageToDAPFromSurendra = hiMessageToDAP("Surendra");

File: higher-order-functions/Currying/SendMessage.js
var sendMessage = function(from, to, text) {
    return '@' + to + ': ' + text + ' - ' + from;
};

sendMessage('Surendra','DAP Team','Hi to all');


var mailCurry = function(from) {
    return function(to) {
        return function(text){
            return '@' + to + ': ' + text + ' - ' + from;
        };
    };
};

var mailFromSurendra = mailCurry('Surendra');
var mailFromSurendraToDAP = mailFromSurendra('DAP Team');
mailFromSurendraToDAP('Hi to all');

mailCurry('Surendra')('DAP Team')('Hi to all');